### PR TITLE
fix: return Left on HttpException instead of throwing

### LINKED ISSUE #48: `Kafka.Avro.SchemaRegistry.getVersions` throws a `HttpException` if the subject does not exist
When querying `Kafka.Avro.SchemaRegistry.getVersions` for a subject, it has a chance to throw a `HttpException` if the subject does not (yet) exist in the schema registry.
This to me seems like inconsistent behaviour. I would expect it to handle the exception (thrown by `Wreq.getWith`), and to return a `Left (SchemaRegistrySubjectNotFound Subject)`, analogous to `SchemaRegistrySchemaNotFound`.

### PR DESCRIPTION
**Pull request by MrcJkb**
* Fixes #48 (and some more)
* Builds on the fixes applied in #45 and #49

This should ensure no more Exceptions are thrown,
but `SchemaRegistryError`s are returned instead.

### CODE REVIEW

**Comment by MrcJkb** [comment]
@newhoggy @tmcgilchrist @nkbt @AlexeyRaga 
I hope you don't mind my pinging you personally (it looks like the PRs I'm basing this one on got missed).

**Comment by AlexeyRaga** [comment]
@MrcJkb thanks for pinging, will have a look soon!

**Comment by MrcJkb** [comment]
Those CI failures don't look related to my PR. Any idea what's going on?

File: src/Kafka/Avro/SchemaRegistry.hs [hunk 1]
```diff
@@ -223,10 +235,22 @@ wrapError someErr = case fromException someErr of
   Just httpErr -> fromHttpError httpErr (\_ -> SchemaRegistrySendError (displayException someErr))
```
Comment (MrcJkb) [inline src/Kafka/Avro/SchemaRegistry.hs:235]: Not directly related to this PR, but catching and wrapping every possible `Exception` could be dangerous.

Comment (AlexeyRaga) [reply src/Kafka/Avro/SchemaRegistry.hs:235]: Yeah, we should only be catching Http/Net exceptions and not the others...

**Review by AlexeyRaga — APPROVED** [review]

**Pull request merged**